spec/lib/manageiq/graphql/types/vm_spec.rb

```diff
@@ -0,0 +1,27 @@
+RSpec.describe ManageIQ::GraphQL::Types::Vm do
+  example "fields" do
+    actual = described_class.fields.keys
+
+    expected = %w(
+      boot_time
+      cloud
+      created_on
+      ems_ref
+      guid
+      id
+      last_perf_capture_on
+      location
+      name
+      power_state
+      previous_state
+      raw_power_state
+      service
+      state_changed_on
+      type
+      uid_ems
+      updated_on
+      vendor
+    )
+    expect(actual).to match_array(expected)
```

Comment:
Honestly, I don't know if this test has a lot of value - it's testing implementation rather than behavior, and has a bit of "testing the framework" feel. But I can see more of a need to test at this level with fields that might have, custom resolvers for instance.

Reply:
Yeah....I think perhaps now that I've had a chance to clarify my thoughts...this i left in as an example of "how to test/interact with the internal API at this level", but not so much as an example of a good thing to test. I'll continue to work on and refine the latter (if you're OK with this for now) as we build things out.